packages/synapse-react-client/src/components/SynapseChat/AccessLevelMenu.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react'
+import { Box, Button, Menu, MenuItem, Typography } from '@mui/material'
+import { AgentAccessLevel } from '@sage-bionetworks/synapse-types'
+import { ArrowDropDown } from '@mui/icons-material'
+
+export type AccessLevelMenuProps = {
+  initAccessLevel: AgentAccessLevel
+  onChange: (newAccessLevel: AgentAccessLevel) => void
+}
+
+const accessLevelOptions = [
+  { value: AgentAccessLevel.PUBLICLY_ACCESSIBLE, label: 'Public Data Only' },
+  {
+    value: AgentAccessLevel.READ_YOUR_PRIVATE_DATA,
+    label: 'Read Your Private Data',
+  },
+  {
+    value: AgentAccessLevel.WRITE_YOUR_PRIVATE_DATA,
+    label: 'Read & Write Your Data',
+  },
+]
+
+export const AccessLevelMenu: React.FunctionComponent<AccessLevelMenuProps> = ({
+  onChange,
+  initAccessLevel,
+}) => {
+  const initOption = accessLevelOptions.find(v => v.value == initAccessLevel)
+  const [selectedAccessLevel, setSelectedAccessLevel] = useState(initOption)
+  const [anchorEl, setAnchorEl] = useState<
+    (EventTarget & HTMLButtonElement) | null
+  >(null)
+  const open = Boolean(anchorEl)
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+
+  return (
+    <Box
+      sx={{
+        display: 'grid',
+        gridTemplateColumns: 'auto 65px 240px',
+        mt: '20px',
+      }}
+    >
+      <Typography variant="label" sx={{ gridColumn: '2', mt: '10px' }}>
+        Access:
+      </Typography>
+      <Button
```

Comment:
We have an existing `DropdownSelect` component that abstracts some of this away, so you may want to look at using that

Reply:
Done!